stonesoup/sensor/sensor.py

```diff
@@ -0,0 +1,48 @@
+from abc import ABC
+
+from ..platform.base import FixedPlatform
+from .base import BaseSensor
+from ..types.array import StateVector
+from ..types.state import State
+
+
+class Sensor(BaseSensor, ABC):
+    """Sensor Base class for general use.
+
+    Most properties and methods are inherited from :class:`~.BaseSensor`, but this class includes
+    crucial functionality and so should be used in preference.
+
+    All sensors must be mounted on a platform to calculate their position and orientation. To
+    make this easier, if the sensor has a position and/or orientation specified in the constructor,
+    and no :attr:`platform_system`, then the default is to create an internally held "private"
+    platform for the Sensor. This restricts the later setting of the :attr:`platform_system` but
+    does allow the Sensor to control (and set) its own position and orientation.
```

Comment:
I think these should be `Property`s (defaulting to `None`), as this can't be serialised into configuration file.

Reply:
Which things should be `Property`s? Position and orientation? These are never saved simply used to initialise a platform, so it seems odd to make them `Property`s. Or maybe I misunderstood?

Reply:
After our discussion this morning, I accept this is an issue, and have added a note to the docs and a warning to the serialisation code, but fixing this properly seems pretty painful. As it is a edge case, we are probably OK with a warning, but if I have a brainwave on how to do it, I will update the code in a later commit. Not promising anything!

Reply:
OK. I think I fixed it up. The problem I was having before was trying to test it too generically, but the simple case is much easier and works now.